.github/workflows/run-profiling.yaml

```
@@ -9,39 +9,106 @@ name: Run profiling
 
 on:
   workflow_dispatch:
+  # Allow profiling to be triggered by comments on pull requests
+  # Trigger is /run profiling
+  issue_comment:
+    types:
+      - created
+  # Profile the model every Saturday at 00:00, 
+  # on the HEAD of master
+  schedule:
+    - cron: 0 0 * * 6
+
+concurrency:
+  # Prevent multiple profiling runs from happening on the same
+  # commit simultaneously.
+  # This ensures we don't overwrite the artifact to be uploaded
+  # with another run that uses the same name, before the results
+  # can be processed. 
+  group: profiling-${{ github.workflow }}-${{ github.ref }}
 
 jobs:
-  run-profiling:
-    name: Setup developer environment
+  set-variables:
+    name: Create unique output file identifier and artifact name
+    runs-on: ubuntu-latest
+    outputs:
+      profiling-filename: ${{ steps.set-output-id.outputs.name }}
+      artifact-name: $${{ steps.set-artifact-name.outputs.name }}
+    steps:
+      - id: set-output-id
```

```suggestion
      profiling-filename: ${{ steps.set-profiling-filename.outputs.name }}
      artifact-name: $${{ steps.set-artifact-name.outputs.name }}
    steps:
      - id: set-profiling-filename
```
Unnecessary but just to keep things consistent